src/ape/api/providers.py

```diff
@@ -166,3 +166,17 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
     @abstractmethod
     def get_events(self, **filter_params) -> Iterator[dict]:
         ...
+
+
+class TestProviderAPI(ProviderAPI):
+    @abstractmethod
+    def snapshot(self) -> Dict:
+        ...
+
+    @abstractmethod
+    def revert(self, snapshot: Dict):
```

Comment:
Using `Dict` because `EthTester` require `dicts`.
And for `hardhat` or `ganache`, provider plugins can just return `{"snapshot_id": 123}` or something.